RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterParamsESProducer.cc

```diff
@@ -0,0 +1,248 @@
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ESProducer.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+#include "HeterogeneousCore/CUDACore/interface/JobConfigurationGPURecord.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusterParamsDeviceCollection.h"
+
+#include <array>
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class PFClusterParamsESProducer : public ESProducer {
+  public:
+    PFClusterParamsESProducer(edm::ParameterSet const& iConfig) : ESProducer(iConfig) {
+      product = std::make_shared<reco::PFClusterParamsHostCollection>(7, cms::alpakatools::host());
```

Comment:
what is `7` here ?

Reply:
if it's the maximum between the barrel and endcap depths, I'd suggest moving it after those declarations and changing it to
```suggestion
      product = std::make_shared<reco::PFClusterParamsHostCollection>(std::max(kMaxDepth_barrel, kMaxDepth_endcap), cms::alpakatools::host());
```